v2/spiffetls/validate.go

```diff
@@ -0,0 +1,39 @@
+package spiffetls
+
+import (
+	"crypto/x509"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+// Validator is a callback used to verify a SPIFFE ID provided in an X509-SVID
+// during the TLS handshake.
+type Validator func(id spiffeid.ID, verifiedChains [][]*x509.Certificate) error
```

Comment:
That's an interesting point. Using a client X509-SVID to authenticate with a Web PKI server feels a little uncommon? If not, we can probably accomodate for that, although I'd imagine the Validator stays the same since I'd imagine normal web PKI hostname verification would be sufficient.